src/Mod/TechDraw/Gui/CommandExtensionDims.cpp

```diff
@@ -0,0 +1,2231 @@
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+# include <QMessageBox>
+# include <iostream>
+# include <string>
+# include <sstream>
+# include <cstdlib>
+# include <exception>
+#endif  //#ifndef _PreComp_
+
+#include <QGraphicsView>
+
+# include <App/DocumentObject.h>
+# include <Base/Exception.h>
+#include <Base/Console.h>
+#include <Base/Type.h>
+# include <Gui/Action.h>
+# include <Gui/Application.h>
+# include <Gui/BitmapFactory.h>
+# include <Gui/Command.h>
+# include <Gui/Control.h>
+# include <Gui/Document.h>
+# include <Gui/Selection.h>
+# include <Gui/MainWindow.h>
+# include <Gui/FileDialog.h>
+# include <Gui/ViewProvider.h>
+
+# include <Mod/Part/App/PartFeature.h>
+
+# include <Mod/TechDraw/App/DrawViewPart.h>
+# include <Mod/TechDraw/App/DrawProjGroupItem.h>
+# include <Mod/TechDraw/App/DrawProjGroup.h>
+# include <Mod/TechDraw/App/DrawViewDimension.h>
+# include <Mod/TechDraw/App/DrawDimHelper.h>
+# include <Mod/TechDraw/App/LandmarkDimension.h>
+# include <Mod/TechDraw/App/DrawPage.h>
+# include <Mod/TechDraw/App/DrawUtil.h>
+# include <Mod/TechDraw/App/Geometry.h>
+
+#include <Mod/TechDraw/Gui/QGVPage.h> //needed ?
+
+
+#include "DrawGuiUtil.h"
+#include "MDIViewPage.h"
+#include "ViewProviderPage.h"
+#include "TaskLinkDim.h"
+
+#include "TaskSelectLineAttributes.h"
+
+/////////////////////////////
+#include <Mod/TechDraw/App/DrawViewSection.h>  // needed
+#include <Mod/TechDraw/App/DrawProjGroupItem.h>
+/////////////////////////////
+
+using namespace TechDrawGui;
+using namespace TechDraw;
+using namespace std;
+
+//internal structures and sort functions
+struct dimVertex{
```

Comment:
The helper classes and functions should be moved to either TechDrawGui or anonymous namespace. This is to avoid possible name collisions of functions in other modules.